pkg/controller/nodes/subworkflow/launchplan.go

```diff
@@ -98,15 +98,15 @@ func (l *launchPlanHandler) CheckLaunchPlanStatus(ctx context.Context, nCtx hand
 	case core.WorkflowExecution_ABORTED:
 		wErr = fmt.Errorf("launchplan execution aborted")
 		err = errors.Wrapf(errors.RemoteChildWorkflowExecutionFailed, nCtx.NodeID(), wErr, "launchplan [%s] failed", childID.Name)
-		return handler.DoTransition(handler.TransitionTypeEphemeral, handler.PhaseInfoFailure(errors.RemoteChildWorkflowExecutionFailed, err.Error(), &handler.ExecutionInfo{
+		return handler.DoTransition(handler.TransitionTypeEphemeral, handler.PhaseInfoFailure(core.ExecutionError_USER, errors.RemoteChildWorkflowExecutionFailed, err.Error(), &handler.ExecutionInfo{
 			WorkflowNodeInfo: &handler.WorkflowNodeInfo{LaunchedWorkflowID: childID},
 		})), nil
 	case core.WorkflowExecution_FAILED:
 		errMsg := fmt.Sprintf("launchplan execution failed without explicit error")
 		if wfStatusClosure.GetError() != nil {
 			errMsg = fmt.Sprintf(" errorCode[%s]: %s", wfStatusClosure.GetError().Code, wfStatusClosure.GetError().Message)
 		}
-		return handler.DoTransition(handler.TransitionTypeEphemeral, handler.PhaseInfoFailure(errors.RemoteChildWorkflowExecutionFailed, errMsg, &handler.ExecutionInfo{
+		return handler.DoTransition(handler.TransitionTypeEphemeral, handler.PhaseInfoFailure(core.ExecutionError_USER, errors.RemoteChildWorkflowExecutionFailed, errMsg, &handler.ExecutionInfo{
```

Comment:
The execution closure error has kind. should we use the kind from there? If it failed due to a system error, should this execution also fail with a system error?

Reply:
yes we should